src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/CryptographyManager.cs

```diff
@@ -1,32 +1,95 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System.Text;
+using System;
 using System.Collections.Concurrent;
+using System.Linq;
 using System.Security.Cryptography;
 using System.Security.Cryptography.X509Certificates;
+using System.Text;
 
 namespace Microsoft.Identity.Client.PlatformsCommon.Shared
 {
     internal static class CryptographyManager
     {
-        private static readonly ConcurrentDictionary<string, RSA> s_certificateToRsaMap = new ConcurrentDictionary<string, RSA>();
+        private static readonly ConcurrentDictionary<string, AsymmetricAlgorithm> s_certificateToRsaMap = new ConcurrentDictionary<string, AsymmetricAlgorithm>();
         private static readonly int s_maximumMapSize = 1000;
 
         public static byte[] SignWithCertificate(string message, X509Certificate2 certificate)
         {
-            if (!s_certificateToRsaMap.TryGetValue(certificate.Thumbprint, out RSA rsa))
+            var chain = new X509Chain
+            {
+                ChainPolicy = BuildChainPolicyChained(),
+            };
+
+            var certificateIsValid = chain.Build(certificate);
+            if (!certificateIsValid)
+            {
+                var chainErrors = string.Join(", ", chain.ChainStatus.Select(e => e.Status.ToString()));
+                throw new InvalidOperationException($"An invalid certificate was supplied. {chainErrors}");
+            }
+
+            chain.Dispose();
+
+            if (!s_certificateToRsaMap.TryGetValue(certificate.Thumbprint, out AsymmetricAlgorithm alg))
             {
                 if (s_certificateToRsaMap.Count >= s_maximumMapSize)
+                {
                     s_certificateToRsaMap.Clear();
+                }
+
+#if NET5_0_OR_GREATER
```

Comment:
MSAL supports the following tfsm: net45, net461, netcore21 and net5-windows10.0.17663.0 
I'd really to understand where all these certs are available. Can you help? May @GeoK can also help.

|   |   RSA|  ECD  | DSA  |   |
|---|---|---|---|---|
| net45  |   |   |   |   |
|  net461 |   |   |   |   |
|  netcore2.1 |   |   |   |   |
|  netstandard13 |   |   |   |   |
|  net5-windows10.0.17663.0 |   |   |   |   |

Reply:
ifdefs are not looking right (e.g. ln 28).
GetECDsaPrivateKey is available on net461, but not on net45.
https://apisof.net/catalog/d72e8acb-f5b9-1f0b-2c55-fe234b01e17b